server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1799,6 +1799,18 @@ boolean shouldRollTranslogGeneration() {
     }
 
     public void onSettingsChanged() {
+        var newEngineFactory = IndicesService.getEngineFactory(indexSettings, engineFactoryProviders);
+        if (newEngineFactory.getClass() != engineFactory.getClass()) {
```

Comment:
No really great to detect a new engine factory. Another option would be to use a setting for defining the to-be-used engine (factory). Or any other suggestion?

Reply:
I think it's fine to detect it like this.